### PR TITLE
Add gzip support as proposed in issue #68

### PR DESCRIPTION
**Pull request by renatoalencar**
Just let gzip functions of zlib read files. When in gzip format these functions decompress files, if not in gzip format, just read as decompressed stream.

### CODE REVIEW

**Comment by vanhauser-thc** [comment]
Hi,
sorry, I am ill at the moment ...
If I understand the patch correctly, instead of normal files it only supports gzip'ed files now?
Or does the gzopen detect if its a normal file or gziped one?
The libz manual is not clear about that.

**Comment by vanhauser-thc** [comment]
ah, sorry, now I see your description text above :)
as I said, I am ill at the moment with fever :)
sorry, to be sure, with the patch it does not matter if the files are gziped or not and it will work?

**Comment by renatoalencar** [comment]
"gzopen can be used to read a file wich is not in gzip format; in this case gzread will directly read from the file without decompression. When reading, this will be detected automatically by looking for the magic two-byte gzip header."

**Comment by renatoalencar** [comment]
This is what i found about when i was reading the documentation, it can be found in zlib.h. gzopen reads the first bytes in stream and detect if it's 1F8B. Thanks for feedbacks :)

**Comment by vanhauser-thc** [comment]
OK, let me rephrase that - have you actually tried that both work, gzip'ed and normal text file?
:)

**Comment by renatoalencar** [comment]
Yes, i used with a wordlist that comes with Kali Linux in /usr/share/wordlists/rockyou.txt.gz, gzip'ed and uncompressed, it worked right at the moment.

**Comment by vanhauser-thc** [comment]
thanks!
